test/e2e/thanos_querier_test.go

```diff
@@ -91,3 +92,137 @@ func TestThanosQueryCanQueryWatchdogAlert(t *testing.T) {
 		},
 	)
 }
+
+func TestMonitoringApiRoles(t *testing.T) {
+	const roleName = "cluster-monitoring-metrics-api"
+	const clusterRoleName = "cluster-monitoring-view"
+	for _, tc := range []scenario{
+		{
+			name:      fmt.Sprintf("assert %s role exists", roleName),
+			assertion: f.AssertRoleExists("cluster-monitoring-metrics-api", f.Ns),
+		},
+		{
+			name:      fmt.Sprintf("assert %s cluster role exists", clusterRoleName),
+			assertion: f.AssertClusterRoleExists(clusterRoleName),
+		},
+		{
+			name:      fmt.Sprintf("thanos querier API is accessible by cluster role %s", clusterRoleName),
+			assertion: testClusterRoleAccessToThanosQuerier,
+		},
+		{
+			name:      fmt.Sprintf("thanos querier API is accessible by role %s", roleName),
+			assertion: testRoleAccessToThanosQuerier,
+		},
+		{
+			name:      "thanos querier API is inaccessible without either role or cluster role",
+			assertion: testBlockedAccessToThanosQuerier,
+		},
+	} {
+		t.Run(tc.name, tc.assertion)
+	}
+}
+
+func testClusterRoleAccessToThanosQuerier(t *testing.T) {
+	const clusterRoleName = "cluster-monitoring-view"
+	const saName = "e2e-thanos-querier"
+
+	cleanUpFuncs := []framework.CleanUpFunc{}
+	cf, err := f.CreateServiceAccount(saName, f.Ns)
```

Comment:
I would create the service account in a dummy namespace, to mimic what end-users should do.